microbenchmarks/singledispatch.py

```diff
@@ -0,0 +1,53 @@
+from functools import singledispatch
+
+from benchmarking import benchmark
+
+NUM_ITER = 100_000
+
+
+class Tree:
+    pass
+
+
+class Leaf(Tree):
+    pass
+
+
+class Node(Tree):
+    def __init__(self, value: int, left: Tree, right: Tree) -> None:
+        self.value = value
+        self.left = left
+        self.right = right
+
+
+@singledispatch
+def calc_sum(x: Tree) -> int:
+    raise TypeError("invalid type for x")
+
+
+@calc_sum.register
+def sum_leaf(x: Leaf) -> int:
+    return 0
+
+
+@calc_sum.register
+def sum_node(x: Node) -> int:
+    return x.value + calc_sum(x.left) + calc_sum(x.right)
+
+
+def build(n: int) -> Tree:
+    if n == 0:
+        return Leaf()
+    return Node(n, build(n - 1), build(n - 1))
+
+
+@benchmark
+def sum_tree():
```

Comment:
Maybe rename to `sum_tree_singledispatch`, for example, to make it easier to see that is mostly about benchmarking `@singledispatch`?

Reply:
Fixed in 5457dc52e8b0fb11173c67e4f059ea12a4b046e2